packages/core-components/src/components/table/table-header/table-header.tsx

```diff
@@ -108,20 +119,34 @@ export class TableHeaderComponent {
             tabIndex={0}
             class={{
               'b2b-table-header__heading': true,
+              [`b2b-table-header__heading--${this.contentAlign}`]: true,
               'b2b-table-header__sort--active': this.active,
             }}
             onClick={this.changeSortDirection}
             onKeyDown={this.changeSortDirection}
             onMouseLeave={this.setInactive}
             onBlur={this.setInactive}>
+            {this.sortIconAlign === SortIconAlignment.LEFT && (
+              <span
+                class={{
+                  'b2b-table-header__sort': true,
+                  [`b2b-table-header__sort--${this.sortDirection}`]: true,
+                  [`b2b-table-header__sort--${this.sortIconAlign}`]: true,
+                }}>
+                {this.sortIcon}
+              </span>
+            )}
             <slot></slot>
-            <span
-              class={{
-                'b2b-table-header__sort': true,
-                [`b2b-table-header__sort--${this.sortDirection}`]: true,
-              }}>
-              {this.sortIcon}
-            </span>
+            {this.sortIconAlign === SortIconAlignment.RIGHT && (
+              <span
+                class={{
+                  'b2b-table-header__sort': true,
+                  [`b2b-table-header__sort--${this.sortDirection}`]: true,
+                  [`b2b-table-header__sort--${this.sortIconAlign}`]: true,
+                }}>
+                {this.sortIcon}
+              </span>
+            )}
```

Comment:
did you try wrapping the whole header content in a flex container that changes direction depending on the sortIconAlignment? Should save you a rerender. Probably not a big save as I don't think alignment is going to change in a component instance, but that way it's still a bit cleaner.